detect_secrets/plugins/base.py

```diff
@@ -145,7 +145,12 @@ def denylist(self) -> Iterable[Pattern]:
     def analyze_string(self, string: str) -> Generator[str, None, None]:
         for regex in self.denylist:
             for match in regex.findall(string):
-                yield match
+                if isinstance(match, tuple):
+                    for submatch in filter(len, tuple):
```

Comment:
```suggestion
                    for submatch in filter(bool, tuple):
```

I think this is what you were going for? Do let me know if it isn't. I think this makes a little more sense in terms of readability.

Reply:
Hello! Thanks! Yes, I think, this more suitable change.

Reply:
Great. Once you merge this fix in, @KevinHock can help push this through to master.

Reply:
Done.